lib/use_packwerk.rb

```diff
@@ -245,4 +245,40 @@ def self.get_offenses_for_files_by_package(files)
     Private::PackwerkWrapper.packwerk_cli_execute_safely(argv, formatter)
     formatter.aggregated_offenses.compact
   end
+
+  sig { void }
+  def self.lint_deprecated_references!
+    contents_before = Private.get_deprecated_references_contents
+    UsePackwerk.execute(['update-deprecations'])
+    contents_after = Private.get_deprecated_references_contents
+    diff = Private.diff_deprecated_references_yml(contents_before, contents_after)
+
+    if diff == ''
+      # No diff generated by `update-deprecations`
+      exit 0
+    else
+      output = <<~OUTPUT
+        All `deprecated_references.yml` files must be up-to-date and that no diff is generated when running `bin/packwerk update-deprecations`.
+        This helps ensure a high quality signal in other engineers' PRs when inspecting new violations by ensuring there are no unrelated changes.
+
+        There are three main reasons there may be a diff:
+        1) Most likely, you may have stale violations, meaning there are old violations that no longer apply.
+        2) You may have some sort of auto-formatter set up somewhere (e.g. something that reformats YML files) that is, for example, changing double quotes to single quotes. Ensure this is turned off for these auto-generated files.
+        3) You may have edited these files manually. It's recommended to use the `bin/packwerk update-deprecations` command to make changes to `deprecated_references.yml` files.
+
+        In all cases, you can run `bin/packwerk update-deprecations` to update these files.
+
+        Here is the diff generated after running `update-deprecations`:
+        ```
+        #{diff}
+        ```
+
+      OUTPUT
+
+      puts output
+      UsePackwerk.config.on_deprecated_references_lint_failure.call(output)
```

Comment:
At Gusto, we can configure this like this:
```
--- /dev/null
+++ b/config/use_packwerk.rb
@@ -0,0 +1,7 @@
+require 'gusto_packwerk'
+
+module GustoPackwerk
+  UsePackwerk.configure do |config|
+    config.on_deprecated_references_lint_failure = -> (output) { Private.post_buildkite_annotation(output) }
+  end
+end
```

I don't *love* this exactly... some alternative ideas were to have this method return a `Result` type with the output, and we could just do whatever we want with that result. However, I liked the idea that we can use the vanilla bin stub (`bin/packs lint_deprecated_references`) without needing to orchestrate anything else.